pkg/controllers/user/logging/utils/configmap.go

```diff
@@ -40,20 +40,19 @@ func UpdateConfigMap(configPath, loggingName, level string, configmaps rv1.Confi
 	if err != nil {
 		return errors.Wrap(err, "buildConfigMap failed")
 	}
-	existConfig, err := configmaps.Controller().Lister().Get(loggingconfig.LoggingNamespace, loggingName)
+	existConfig, err := configmaps.Get(loggingName, metav1.GetOptions{})
 	if err != nil {
 		if apierrors.IsNotFound(err) {
-			if _, err = configmaps.Create(configMap); err != nil && !apierrors.IsAlreadyExists(err) {
+			if existConfig, err = configmaps.Create(configMap); err != nil && !apierrors.IsAlreadyExists(err) {
 				return errors.Wrap(err, "create configmap failed")
 			}
-			return nil
+		} else {
+			return errors.Wrap(err, "get configmap failed")
 		}
-		return errors.Wrap(err, "list configmap failed")
 	}
-	newConfigMap := existConfig.DeepCopy()
-	newConfigMap.Data = configMap.Data
-	if _, err = configmaps.Update(newConfigMap); err != nil {
-		errors.Wrap(err, "update configmap failed")
+	existConfig.Data = configMap.Data
+	if _, err = configmaps.Update(existConfig); err != nil {
+		return errors.Wrap(err, "update configmap failed")
```

Comment:
we perhaps need to add in the log which configmap failed; the message is too generic, and won't help user to figure out the destination configmap